### PR TITLE
Drop rc dependency

### PR DESCRIPTION
**Pull request by springmeyer**
This dependency is barely used and it is easy to get by without it. Addresses #535

However this removes the s3 host env support just because it was using rc. Not sure if that is absolutely critical to keep or not. /cc @bmacnaughton

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
-- Added back support for symlinks inside the tarball
+- New: support for staging and production s3 targets (see README.md)
+- BREAKING: no longer supporting `node_pre_gyp_accessKeyId` & `node_pre_gyp_secretAccessKey`, use `AWS_ACCESS_KEY_ID` & `AWS_SECRET_ACCESS_KEY` instead to authenticate against s3 for `info`, `publish`, and `unpublish` commands.
 - Dropped node v6 support, added node v14 support
 - Switched tests to use mapbox-owned bucket for testing
 - Added coverage tracking and linting with eslint
+- Added back support for symlinks inside the tarball
 - Upgraded all test apps to N-API/node-addon-api
-- New: support for staging and production s3 targets (see README.md)
-- Added `node_pre_gyp_s3_host` env var which has priority over the `--s3_host` option or default.
```
Comment (bmacnaughton) [inline CHANGELOG.md:11]: i wasn't thinking get rid of the env var completely; it's still important to override the default without modifying the command line(s) in package.json.

is a new name preferred?

File: lib/node-pre-gyp.js [hunk 2]
```diff
@@ -255,13 +254,7 @@ proto.setBinaryHostProperty = function(command) {
   if (command === 'publish') {
     target = 'staging_host';
   }
-  // each test is separate as opposed to combined like "if (this.rc[x] || this.opts[x])" to give
-  // priority to env var settings over command line.
-  if (this.rc['s3_host'] === 'staging') {
-    target = 'staging_host';
-  } else if (this.rc['s3_host'] === 'production') {
-    target = 'production_host';
-  } else if (this.opts['s3_host'] === 'staging') {
```
Comment (bmacnaughton) [inline lib/node-pre-gyp.js:264]: related to previous point; i'll plan on a commit that just checks the env var in this section.

**Review by bmacnaughton — APPROVED** [review]
looks good other than removing the note about the env var from the doc/changelog.

File: lib/node-pre-gyp.js (hunk 2 above)
Comment (springmeyer) [reply lib/node-pre-gyp.js:264]: 👍  per email, please add as a followup. So I'll merge this as-is and let you fix up with `process.env` usage as you see fit (I don't have opinions on the naming)

**Pull request merged**